### DISCUSSION TITLE
Toggle component

### DISCUSSION THREAD

**Discussion by joaorbp** [dxc-technology/halstack-style-guide#156] [Ideas] [answered]
Hello, some feedback and suggestions for the toggle component.

## From documentation

From the documentation this is how the toggle component should look like:

![image](https://user-images.githubusercontent.com/1644143/117001010-c05af800-acd9-11eb-9aae-4f1a8347a2df.png)

In my opinion I think the toggle component should just be a group of buttons. 

Based on the current documentation this is how the buttons look like:

![image](https://user-images.githubusercontent.com/1644143/117001129-e1bbe400-acd9-11eb-8ab3-af6b8c0feb0e.png)

I think for consistency it would be better for the toggle to just use the current buttons design, but just group them together.

![image](https://user-images.githubusercontent.com/1644143/117001261-0748ed80-acda-11eb-97b1-005d0de6758b.png)

We could still have the same basic and outline options, being the basic the primary buttons and the outlined the secondary buttons.

The toggle component also seems to use the ripple effect, I think a decision would need to be made, for all the buttons to use the ripple effect or none of the buttons to use the effect, for consistency in the Halstack design . 

![image](https://user-images.githubusercontent.com/1644143/117002227-3449d000-acdb-11eb-9394-b06bfc5b78f4.png)

Also the color yellow is introduced on the toggle component, when is not part of the Halstack colours.

![image](https://user-images.githubusercontent.com/1644143/117001977-e59c3600-acda-11eb-87a5-baa0560c466a.png)

![image](https://user-images.githubusercontent.com/1644143/117002023-f2208e80-acda-11eb-8082-59b8c48fa535.png)

So far I can't see any other buttons in the Halstack containing yellow.

Thank you for taking the time to read, hopefully this will spark some discussion.

**Comment by aweell** [comment]
Thanks @joaorbp for opening the discussion!

Documentation of the toggle group hasn't been yet updated. The last component styles are available in:
https://developer.dxc.com/tools/react/next/#/components/toggleGroup

Apart from that:

**Semantically speaking**

> In my opinion I think the toggle component should just be a group of buttons.

Not only related to how is built but also how is named:`button group` or `split button` seems more convenient than `toggle group`. 

Merriam-Webster defines the toggle as:

> **Toggle** _Noun_
> 
> computers : a setting that can be switched between two different options by performing a single action (such as selecting a menu option or pressing a key)

Since the current `toggle group` allows multiple selections and there is another component named [`switch`](https://developer.dxc.com/design/components/switch) (mainly known as a toggle) that behaves exactly like its defined, the name should be reviewed.

**Structure**

Since I propose the idea of changing the name to `button group`, let me add some questions to the discussion supposing it's based mainly on buttons :wink:

* Apart from the solution you present (segmented group) could be interesting to present the buttons independently? Shopify [Polaris](https://polaris.shopify.com/components/actions/button-group) example
* Should be possible to provide a more complex selection by adding a dropdown button? Uber [Base design system](https://baseweb.design/components/button-group/) example
* Should be provided different spacing/stacking methods? Adobe [Spectrum design system](https://opensource.adobe.com/spectrum-css/buttongroup.html) example
* Would be interesting the exploration of different `button group` sizes? [Bootstrap](https://getbootstrap.com/docs/4.3/components/button-group/) example
* Should be allowed to combine different button variants? [Seek styleguides](https://seek-oss.github.io/seek-style-guide/button-group/)

**Comment by aweell** [comment, marked as answer]
Hi @joaorbp,

The toggle documentation has been updated in release 3.8.0! 

You can check the toggle documentation [here](https://github.com/dxc-technology/halstack-style-guide/tree/master/guidelines/components/toggle). 

Or see [what we updated](https://github.com/dxc-technology/halstack-style-guide/releases/tag/3.8.0) from the last release.

**Reply by joaorbp** [reply]
They look good, great work.
